Create Itohan's Trinket Post (lambda and sorting)

itohosagie opened the pull request.

itohosagie committed:
Create Itohan's Trinket Post (lambda and sorting)

benderliz reviewed (commented):
Very thoughtful of you to attempt to use lambda! I wish I knew how to use it to resolve the issue.

Elizard0926 reviewed (commented):
I was also stumped with how to use lambda when we first encountered it.

itohosagie committed:
Revising reflection portion (adding more detail)

Elaborating a bit more on my approach to this exercise within the textbook, as well as existing challenges.

MishaCienfuegos reviewed (approved):
The markdown formatting and header looks great. Nice work!
